Ninjathulhu/Ninjathulhu.Game/Entity.cs

```diff
@@ -3,80 +3,75 @@
 
 namespace Ninjathulhu.Game
 {
-    class EntityComponents
+    public class EntityComponents
     {
-        private Entity Entity;
+        private readonly Entity _entity;
 
-        private Dictionary<Type, Component> Components = new Dictionary<Type, Component>();
+        private readonly Dictionary<Type, Component> _components = new Dictionary<Type, Component>();
 
         public EntityComponents(Entity entity)
         {
-            Entity = entity;
+            _entity = entity;
         }
 
         public Component Get<TComponent>()
             where TComponent : Component
         {
-            return Components[typeof(TComponent)];
+            return Get(typeof (TComponent));
         }
 
         public Component Get(Type componentType)
         {
-            return Components[componentType];
+            if(_components.ContainsKey(componentType))
+                return _components[componentType];
+            return null;
         }
 
         public Component Attach<TComponent>()
             where TComponent : Component, new()
         {
-            if (Components.ContainsKey(typeof(TComponent)))
-            {
-                return Components[typeof(TComponent)];
-            }
-
-            var newComponent = new TComponent();
-            Components[typeof(TComponent)] = newComponent;
-            newComponent.AttachTo(Entity);
-            return newComponent;
+            return Attach(typeof (TComponent));
         }
 
         public Component Attach(Type type)
         {
-            if (Components.ContainsKey(type))
+            if (_components.ContainsKey(type))
             {
-                return Components[type];
+                return _components[type];
             }
 
-            Component newComponent = Activator.CreateInstance(type) as Component;
+            var newComponent = Activator.CreateInstance(type) as Component;
             if (newComponent == null) { return null; }
```

Comment:
I see, so this returns `null` if for whatever reason we pass in a type that's not a `Component`. I still think I'd rather fail early and throw a `NullReferenceException` in that case; we really shouldn't try to recover from passing a non-`Component` to `EntityComponents.Attach()`.